### PR TITLE
Fix documentation for `--lock-timeout`

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR fixes the documentation for `--lock-timeout` flag after we noticed it was incorrect.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/flags/endtoend/vtbackup.txt [hunk 1]
```diff
@@ -147,7 +147,7 @@ Flags:
       --keep-alive-timeout duration                                 Wait until timeout elapses after a successful backup before shutting down.
       --keep_logs duration                                          keep logs for this long (using ctime) (zero to keep forever)
       --keep_logs_by_mtime duration                                 keep logs for this long (using mtime) (zero to keep forever)
-      --lock-timeout duration                                       Maximum time for which a shard/keyspace lock can be acquired for (default 45s)
+      --lock-timeout duration                                       Maximum time to wait for while acquiring a lock from the topo server (default 45s)
```
Comment (mattlord) [inline go/flags/endtoend/vtbackup.txt:150]: Supper nitty, but small suggestion:
```suggestion
      --lock-timeout duration                                       Maximum time to wait when attempting to acquire a lock from the topo server (default 45s)
```

**Review by mattlord — APPROVED** [review]
Thanks, @GuptaManan100 !

**Review by frouioui — APPROVED** [review]

File: changelog/20.0/20.0.0/summary.md [hunk 2]
```diff
@@ -33,6 +33,7 @@
     - [New `healthcheck-dial-concurrency` flag](#healthcheck-dial-concurrency-flag)
     - [New minimum for `--buffer_min_time_between_failovers`](#buffer_min_time_between_failovers-flag)
     - [New `track-udfs` vtgate flag](#vtgate-track-udfs-flag)
+    - [Documentation fix for `--lock-timeout`](#documentation-lock-timeout)
```
Comment (deepthi) [inline changelog/20.0/20.0.0/summary.md:36]:
```suggestion
    - [Help text fix for `--lock-timeout`](#documentation-lock-timeout)
```

File: changelog/20.0/20.0.0/summary.md [hunk 3]
```diff
@@ -333,6 +334,10 @@ The `--buffer_min_time_between_failovers` `vttablet` flag now has a minimum valu
 
 The new `--track-udfs` flag enables VTGate to track user defined functions for better planning.
 
+#### <a id="documentation-lock-timeout"/>Documentation fix for `--lock-timeout`
+
+The documentation for the flag `--lock-timeout` was noticed to be incorrect. We were documenting it as a flag that controlled the duration for which the shard lock was acquired for. It is actually the duration for which we wait while acquiring a lock from the topology server.
```
Comment (deepthi) [inline changelog/20.0/20.0.0/summary.md:339]:
```suggestion
The help text for the flag `--lock-timeout` was incorrect. We were documenting it as a flag that controlled the duration for which the shard lock was acquired. It is actually the maximum duration for which we wait while attempting to acquire a lock from the topology server.
```

**Review by deepthi — COMMENTED** [review]
You already have 2 approvals :) I do think the suggested changes should be made. We want help text and docs to be as clear as possible.

**Comment by deepthi** [comment]
Moved back to draft until feedback has been addressed.

**Comment by deepthi** [comment]
@GuptaManan100 don't we need to update the website docs for this as well? similar to https://github.com/vitessio/vitess/pull/15919#issuecomment-2140680722

**Pull request merged**